library/src/main/java/com/novoda/downloadmanager/DownloadBatchRequirementRules.java

```diff
@@ -1,32 +1,18 @@
 package com.novoda.downloadmanager;
 
-import java.util.ArrayList;
-import java.util.List;
-
-final class DownloadBatchRequirementRules implements DownloadBatchRequirementRule {
-
-    private final List<DownloadBatchRequirementRule> rules;
-
-    static DownloadBatchRequirementRules newInstance() {
-        return new DownloadBatchRequirementRules(new ArrayList<>());
-    }
-
-    private DownloadBatchRequirementRules(List<DownloadBatchRequirementRule> rules) {
-        this.rules = rules;
-    }
-
-    void addRule(DownloadBatchRequirementRule storageRequirementRule) {
-        rules.add(storageRequirementRule);
-    }
-
-    @Override
-    public boolean hasViolatedRule(DownloadBatchStatus downloadBatchStatus) {
-        for (DownloadBatchRequirementRule requirementRule : rules) {
-            if (requirementRule.hasViolatedRule(downloadBatchStatus)) {
-                return true;
-            }
-        }
-
-        return false;
-    }
+/**
+ * Used to aggregate multiple {@link DownloadBatchRequirementRule}
+ * and expose it as a single {@link DownloadBatchRequirementRule},
+ * allowing calling code to identify which of the component rules was violated, if any.
+ */
+interface DownloadBatchRequirementRules extends DownloadBatchRequirementRule {
```

Comment:
I think we do not need this interface, unless you think that a client should be able to provide a custom implementation of it (that I do not think its useful). If you remove this interface then you'd have a clean `DownloadBatchRequirementRules` class and you won't need this bit

```
/**
     * @return [null] - a composite rule isn't identifiable by itself.
     * Retrieve the code of the actually violated rule (if any) via {@link #getViolatedRule#getCode()}.
     */
    @Nullable
    @Override
    public Integer getCode() {
        return null;
    }
```